lua/lsp-status/statusline.lua

```diff
@@ -97,9 +97,39 @@ local function statusline_lsp(bufnr)
   return symbol .. config.indicator_ok .. ' '
 end
 
+local function statusline_errors(icon, bufh)
+  icon = (icon or '✗') .. ' '
+  bufh = bufh or vim.api.nvim_get_current_buf()
+  local buf_diagnostics = diagnostics(bufh)
+
+  local errors = buf_diagnostics.errors
+  return icon .. errors
+end
+
+local function statusline_warnings(icon, bufh)
```

Comment:
We could reduce some code duplication by making a factory function `make_statusline_component`, parameterized on a fallback icon and diagnostics key, that returns a function of the form of these new functions. This would also make it easier to add future statusline components (e.g. for info).